src/components/SubmitComment.jsx

```diff
@@ -16,10 +16,18 @@ export default function SubmitComment({ id }) {
   const submitComment = () => {
     const commentBody = { username: user, body: newComment };
 
-    postArticleComment(id, commentBody).then((res) => {
-      setIsCommentSuccessful(true);
-      setNewComment('');
-    });
+    postArticleComment(id, commentBody)
+      .then((res) => {
+        setIsCommentSuccessful(true);
+        setNewComment('');
+        setUserHasPostedComment(true);
+      })
+      .then(() => {
+        setTimeout(() => {
+          setIsCommentSuccessful(false);
+          setUserHasPostedComment(false);
+        }, 1000);
```

Comment:
Similar to the last commit, i don't think setTimeout is the optimal logic here - so please suggest how else I can refactor this.